lib/elixir/lib/process.ex

```diff
@@ -282,10 +282,29 @@ defmodule Process do
   Even if the timer had expired and the message was sent, this function does not
   tell you if the timeout message has arrived at its destination yet.
 
+  ## Options
+
+    * `:async` - (boolean) when `false`, the request for cancellation is
+      synchronous. When `true`, the request for cancellation is asynchronous,
+      meaning that the request to cancel the timer is issued and `:ok` is
+      returned right away. Defaults to `false`.
+
+    * `:info` - (boolean) whether to return information about the timer being
+      cancelled. When the `:async` option is `false` and `:info` is `true`, then
+      either an integer or `false` (like described above) is returned. If
+      `:async` is false and `:info` is `false`, `:ok` is returned. If `:async`
+      is `true` and `:info` is `true`, a message in the form `{:cancel_timer,
+      timer_ref, result}` (where `result` is an integer or `false` like
+      described above) is sent to the caller of this function when the
+      cancellation has been performed. If `:async` is `true` and `:info` is
+      `false`, no message is sent.
+
   Inlined by the compiler.
   """
-  @spec cancel_timer(reference) :: non_neg_integer | false
-  defdelegate cancel_timer(timer_ref), to: :erlang
+  @spec cancel_timer(reference, []) :: non_neg_integer | false
+  @spec cancel_timer(reference, nonempty_list(option)) :: non_neg_integer | false | :ok
```

Comment:
I think that splitting `cancel_timer/2` spec is not clear to a human even if it is optimal for a machine. I think it would be clearer to leave `cancel_timer/1`  spec and have a single `cancel_timer/2` spec.